From c58f67d575228b4c133bb7b525c01e8e91114b7f Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke.mehrtens@intel.com>
Date: Fri, 11 Sep 2020 16:53:44 +0200
Subject: [PATCH] PONRTSYS-7999: Request optional SyncE clock

Request the SyncE clock in addition. This is optional and only needed if
SyncE is really done.
---
 drivers/net/ethernet/lantiq/xrx500_phy_fw.c | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/drivers/net/ethernet/lantiq/xrx500_phy_fw.c b/drivers/net/ethernet/lantiq/xrx500_phy_fw.c
index 03a1f7b7fae1..d7f01bd05c22 100644
--- a/drivers/net/ethernet/lantiq/xrx500_phy_fw.c
+++ b/drivers/net/ethernet/lantiq/xrx500_phy_fw.c
@@ -37,6 +37,7 @@ struct xway_gphy_data {
 	struct device *dev;
 	struct regmap *syscfg, *cgu_syscfg, *chipid_syscfg, *aspa_syscfg, *pdi_base, *fcsi_base;
 	struct clk *clk;
+	struct clk *clk_synce;
 
 	dma_addr_t dma_addr;
 	const char *fw_name;
@@ -478,6 +479,24 @@ static int prx300_dt_parse(struct xway_gphy_data *priv)
 		return ret;
 	}
 
+	/* get SyncE clock */
+	priv->clk_synce = devm_clk_get(&pdev->dev, "synce");
+	/* clock is optional */
+	if (priv->clk_synce == ERR_PTR(-ENOENT)) {
+		priv->clk_synce = NULL;
+	} else {
+		if (IS_ERR(priv->clk_synce)) {
+			dev_err(&pdev->dev, "unable to get synce clk\n");
+			return PTR_ERR(priv->clk_synce);
+		}
+
+		ret = clk_prepare_enable(priv->clk_synce);
+		if (ret < 0) {
+			dev_err(&pdev->dev, "failed to enable clock: %d\n", ret);
+			return ret;
+		}
+	}
+
 	/* get chiptop regmap */
 	priv->syscfg = syscon_regmap_lookup_by_phandle(priv->dev->of_node,
 						       "intel,syscon");
